llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/VectorExt/IR/VectorExtAttrs.td

```diff
@@ -99,5 +99,170 @@ def LayoutAttr : IREEVectorExt_Attr<"Layout",
   }];
 }
 
-#endif // IREE_DIALECT_VECTOREXT_ATTRS
+def LayoutV2Attr : IREEVectorExt_Attr<"LayoutV2",
+      [ DeclareAttrInterfaceMethods<VectorLayoutInterface> ]> {
+  let mnemonic = "layoutv2";
+  let summary = [{A layout representing how a GPU thread hierarchy is distributed over a shape}];
+  let description = [{
+    This layout explicitly defines how a shape is distributed over a tile
+    heirarchy. We consider the following levels of hierarchy:
+
+    1. Subgroups per Workgroup
+    2. Threads per Subgroup
+    3. Elements per Thread
+
+    Each level of heirarchy represents a tile over the next level of
+    heirarchy, except the last level, which represents elements. The tiling is
+    done on each dimension of the original vector shape. For example, for a
+    shape <16x16x16>, the tiles are 3-D. For a shape <32x32>, the tiles are
+    2-D.
+
+    We now describe each level of tiling:
+
+    1. Subgroups per Workgroup
+
+    This level of tiling is also known as "warp distribution". It represents
+    how subgroups are distributed over a workgroup. The total number of
+    subgroups are expected to be equal to the workgroup size of the GPU.
+
+    The subgroups are placed contigiously with their shape and ordering
+    determined by:
+      - `subgroups_per_workgroup`: Sizes of this level of tiling
+      - `subgroup_order`: Ordering of dimensions, from outermost to innermost
+
+    For example, subgroups_per_workgroup=[4, 2], subgroup_order=[0, 1] will
+    arrange the subgroups in the order:
+
+    0 4
+    1 5
+    2 6
+    3 7
+
+    2. Threads per Subgroup:
+
+    Threads in a subgroup are distributed in three levels.
+
+    The first level, batches, are a way to represent instruction unrolling. For
+    example, an intrinsic which can only take 4x4 shape at a time, uses batches
+    to unroll a 16x16 shape to the native intrinsice shape.
+
+    Batches can be thought of as loops around the original layout:
+
+    for b_0 in range(batch_0):
+      for b_1 in range(batch_1):
+        ...
+
+    Batches are represented using two attributes:
+      - batches_per_subgroup: Ranges of each loop
+      - batch_order: Ordering of each loop, from outermost to innermost
+
+    The second level, duplicates, are a way to represent thread duplication
```

Comment:
as discussed, thread duplication is misleading. `outer_vector` or `outer_elements` is clearer about what that is trying to map to in the final instruction/intrinsic.